src/redux/books/books.js

```diff
@@ -41,15 +27,59 @@ const bookReducer = (state = initialState, action) => {
   }
 };
 
-export const additem = (id, title, authorname) => ({
+export const additem = (id, title, authorname, category) => ({
   type: ADD_BOOKS,
   id,
   title,
   authorname,
+  category,
 });
 
 export const removeitem = (id) => ({
   type: REMOVE_BOOKS,
   id,
 });
+
+export async function fetchBooks(dispatch) {
+  const response = await fetch(BASE_URL);
+  const data = await response.json();
+  const objValues = Object.entries(data);
+  objValues.forEach((element) => {
+    const bookId = uuidv4();
+    const bookTitle = element[1][0].title;
+    const bookAuthor = element[1][0].author;
+    const bookCategory = element[1][0].category;
+    dispatch(additem(bookId, bookTitle, bookAuthor, bookCategory));
+  });
+}
+
+export const createBook = (id, title, author, category) => async (dispatch) => {
+  const result = await fetch(BASE_URL, {
+    method: 'POST',
+    body: JSON.stringify({
+      item_id: id,
+      title,
+      author,
+      category,
+    }),
+    headers: {
+      'Content-type': 'application/json; charset=UTF-8',
+    },
+  });
+  if (result) {
+    dispatch(additem(id, title, author, category));
+  }
+};
+
+export const removeBook = (id) => async (dispatch) => {
+  const result = await fetch(`${BASE_URL}/${id}`, {
+    method: 'DELETE',
+    headers: {
+      'Content-type': 'application/json; charset=UTF-8',
+    },
+  });
+  if (result) {
+    dispatch(removeitem(id));
+  }
+};
```

Comment:
- [ ] Great job so far! 👏🏻  However, I noticed that your books are not permanently deleted and they still show up when I reload the page. Please correct this